runelite-client/src/main/java/net/runelite/client/ui/overlay/components/ImageComponent.java

```diff
@@ -52,16 +57,26 @@ public Dimension render(Graphics2D graphics)
 			return null;
 		}
 
-		graphics.drawImage(image, preferredLocation.x, preferredLocation.y, null);
+		BufferedImage toRender = scaledImage != null ? scaledImage : image;
+		graphics.drawImage(toRender, preferredLocation.x, preferredLocation.y, null);
 		final Dimension dimension = new Dimension(image.getWidth(), image.getHeight());
 		bounds.setLocation(preferredLocation);
 		bounds.setSize(dimension);
 		return dimension;
 	}
 
 	@Override
-	public void setPreferredSize(Dimension dimension)
+	public void setPreferredSize(Dimension preferredSize)
 	{
-		// Just use image dimensions for now
+		boolean nulled = preferredSize == null || image == null;
+		boolean sameSize = !nulled && (preferredSize.width == image.getWidth() && preferredSize.height == image.getHeight());
+		if (nulled || sameSize)
+		{
+			scaledImage = null;
+		}
+		else
+		{
+			scaledImage = ImageUtil.resizeImage(image, preferredSize.width, preferredSize.height);
```

Comment:
also compare new prefferedSize to old one so you dont create new images when not necessary

Reply:
Good idea, will add.